src/pages/mods/[id].tsx

```diff
@@ -17,6 +17,78 @@ export const getStaticPaths: GetStaticPaths = async () => {
   };
 };
 
+class InvalidGitHubLinkError extends Error {
+  constructor(message?: string) {
+    super(message || "GitHub link must be 'https://github.com/{org}/{repo}' or 'https://github.com/{org}/{repo}/tree/{ref}/{...path}'");
+  }
+}
+
+interface GitHubContentLink {
+  organization: string,
+  repository: string,
+  path: string,
+  /**
+   * commit/branch/tag
+   */
+  ref?: string,
+};
+
+function parseGitHubContentLink(url: string): GitHubContentLink {
+  const githubLink = new URL(url);
+  const hostname = githubLink.hostname;
+  if (hostname !== "github.com") {
+    throw new Error("The given url's host is not 'github.com'.");
+  }
+
+  const pathname = githubLink.pathname;
+  const splat = pathname.split('/').filter(x => x !== '');
```

Comment:
I don't understand, but in what sense is "splat" used?

Reply:
It is my bad English. `split`'s past tense is `split`.